docs/app/monitoring-and-observability/logging-conventions.md

```diff
@@ -0,0 +1,48 @@
+# Logging Conventions
+
+## Purpose
+
+Logging is a valuable tool for engineering teams to support products in production. It enables debugging, real-time alerts, and real-time business metrics. This document is for engineers to have a set of guidelines and rationale for how to approach logging events in our system.
+
+## Principles
+
+### Make code observability a primary tool for debugging and reasoning about production code
+
+When a user runs into an issue in production, logs offer one of the primary ways of understanding what happened. This is especially important for situations where we can’t or don’t know how to reproduce the issue. In general it is not feasible to attach a debugger to production systems, or to set breakpoints and inspect the state of the application in production, so logs offer a way to debug through “print statements”.
+
+### Make it easy for on-call engineers to search for logs in the codebase
+
+While it is important to use logs to reason about the code, it is often equally important to be able to understand what code is being run from the logs. This is especially true for on-call engineers who may encounter production issues that are in areas of the codebase that they are not directly familiar with. Thus it is important to make it easy to find the associated place in the code that triggered logs.
+
+### Optimize for ease, flexibility, and speed of queries over DRY principles
+
+Log querying systems are often limited in their querying abilities. Most log databases are not relational databases, and as such to enable effective querying of logs, we should think about how we design the structure of our log messages and metadata.
+
+## Guidelines
+
+### General
+
+- **Minimize logic or dynamic behavior** – Logs are often used for troubleshooting, so minimize dynamic behavior in logs to reduce the chance that the logging system itself could be contributing to adding complexity to an issue.
+- **Log all key points throughout the application** – Include logs for:
+  - Important branches in business logic (so you can reason from the logs which branch the code took in production without the aid of a debugger)
+  - Before and after actions that are important or actions that have a nontrivial chance of failing (e.g. external service calls, or calls to legacy systems)
+- **Log both successes and failures** – Successes can be useful for debugging what steps were completed successfully to narrow down what failed. In addition, logging successes can be useful for powering real-time business metrics.
+
+### Log event type
+
+- **INFO for user/client error** – Use INFO events if it’s a user or client error, and ERROR events if it’s a system error. This helps us set stricter alerts on system errors that won’t be affected by fluctuations in user behavior.
```

Comment:
I don't know if it's possible to do this for a template, but some guidance on when to use info vs. warning vs error generally would be useful.

E.g., for the payments system on PFML:
 - `warning` has no impact that (vs. `info`)
 - but `error` triggers a low priority alert for the on-call engineer to investigate
That's behavior, so maybe that's project specific, but we've had some back and forth on the team of what the semantics should be. My take is:
 - INFO -- nothing's wrong with the system (though as noted here, the user may have done something erroneously.)
 - WARNING -- something _may_ be wrong with the system, a human should look into this but it's not urgent.
 - ERROR -- something has _definitely_ gone wrong with the system and require human intervention.
 
 But others would say `warning` should be treated as "somewhat out of the ordinary but not necessarily erroneous behavior" that only needs investigation if there are _more_ warnings than usual on a given day. (E.g., usually some process logs 25 warnings, yesterday it logged 200.)

Reply:
@KevinJBoyer thanks for calling out the lack of guidance on when to use warning vs error! agree with the semantics you suggested. i updated this section can you take another look

Reply:
Nice, yeah this makes a ton of sense to me and helps clear up the ambiguity!